pkg/api/pod/util.go

```diff
@@ -809,16 +818,16 @@ func dropDisabledPodStatusFields(podStatus, oldPodStatus *api.PodStatus, podSpec
 		podStatus.HostIPs = nil
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) && !rroInUse(oldPodSpec) {
-		for i := range podStatus.ContainerStatuses {
-			podStatus.ContainerStatuses[i].VolumeMounts = nil
-		}
-		for i := range podStatus.InitContainerStatuses {
-			podStatus.InitContainerStatuses[i].VolumeMounts = nil
-		}
-		for i := range podStatus.EphemeralContainerStatuses {
-			podStatus.EphemeralContainerStatuses[i].VolumeMounts = nil
```

Comment:
Is this change related to this PR?

Reply:
Thanks for the catch! I put this code back🙇

Reply:
addressed in 2db0c01. sorry for the rebase 🙇